openedx/frontend-app-course-authoring
# feat: home studio search filters

### Description

This PR adds a search input and two dropdown menus to filter courses in studio home:

  -  first dropdown menu: Allows us to filter by "All courses," "Active," or "Archived" courses.
  -  second dropdown menus: Allow for ascending/descending course names and the newest/oldest courses.
    
  
**NOTE:** this applies only for "courses" tab

This Uses `ENABLE_HOME_PAGE_COURSE_API_V2` in the environment
  
### DEMO



https://github.com/openedx/frontend-app-course-authoring/assets/134975835/13c02b94-58dc-4929-bcc9-c443d197730b

<img width="786" alt="Screen Shot 2024-04-04 at 8 17 11 AM" src="https://github.com/openedx/frontend-app-course-authoring/assets/134975835/43ed4d33-f61b-44b0-a90a-f544fb92b0a4">


  <img width="336" alt="Screen Shot 2024-02-19 at 8 25 54 PM" src="https://github.com/openedx/frontend-app-course-authoring/assets/134975835/80f2e1af-8385-41ec-b3f3-92af5b60b1d5">
  
<img width="351" alt="Screen Shot 2024-02-19 at 8 26 13 PM" src="https://github.com/openedx/frontend-app-course-authoring/assets/134975835/4b015a60-1eb7-44ad-991e-c0c249a1cc57">


### Supporting information
To review a little bit more about this implementation, please check this:
https://openedx.atlassian.net/wiki/spaces/OEPM/pages/4032856101/Studio+Home+incremental+upgrades+-+product+approach

### How to test it
1. Set up your edx-platform
    ```
     cd path-to-your-edx-platform
     git fetch origin pull/34175/head:MJG/course-home-api-v2-pagination
     git checkout MJG/course-home-api-v2-pagination
    ```
2. Check it out this branch inside frontend-app-course-authoring
    ```
     cd path-to-frontend-app-course-authoring
     git fetch origin pull/846/head:jv/feat-home-studio-filters
     git checkout jv/feat-home-studio-filters
     nvm use && npm install
     npm start
    ```
